AzFileDiagnostics/Linux/AzFileDiagnostics.sh

```diff
@@ -0,0 +1,746 @@
+do
+	case $opt in
+		yes)
+			#comform to BASH, 0 means true.
+			break
+			;;
+		no)
+			exit 0
+			;;
+		*) echo please type 1 or 2;;
+	esac
+done
+
+
+## Prompt user to select diagnostics option
+print_log "Do you want to tun on diagnostics logs"
+
+options=("yes" "no")
+
+select opt in "${options[@]}"
+do
+	case $opt in
+		yes)
+			#comform to BASH, 0 means true.
+			DIAGON=0
+			break
+			;;
+		no)
+			DIAGON=1
+			break
+			;;
+		*) echo please type 1 or 2;;
+	esac
+done
+
+## Prompt user to type the local mount point and storage account access key.
+mountpoint=''
+while [ -z "$mountpoint" ] ; 
+do
+	print_log "type the local mount point, followed by [ENTER]:"
+	read mountpoint
+done
+
+eval mountpoint="$mountpoint"
+
+if [ ! -d "$mountpoint" ] ;then
+	print_log "mount point "$mountpoint" does not exist, create it now" 'info'
+	mkdir -p "$mountpoint"
+fi
+
+password=''
+while [ -z "$password" ] ;
+do
+	print_log "Type the storage account access key, followed by [ENTER]:"
+	read password
+done
+
+password=\'$password\'
+username=$( echo "$SAFQDN" | cut -d '.' -f 1)
+
+if [ "$DIAGON" -eq 0 ]; then
+	enable_log
+fi
+
+
+command="mount -t cifs "$UNCPATH"  "$mountpoint" -o vers=3.0,username="$username",password=$password,dir_mode=0777,file_mode=0777,sec=ntlmssp"
+print_log "Try with mounting share using SMB3.0"
+print_log "$command" "info"
+sudo sh -c "$command" 2>/tmp/mount.error
+
+if [[ $? -gt 0 ]] ;then
+	echo "Fail to mount with SMB3.0"
+
+	grep "Permission denied" /tmp/mount.error >/dev/null 2>&1
+	if [ $? -eq 0 ]; then
+		#SMB 3.0, provilde hints to Cx
+		echo "The Azure file share failed to mount with mount error (13): Permission denied"
+		echo "This could be due to a number of reasons."
+		echo "To resolve this issue, follow the steps in the Azure Files troubleshooting guide: https://docs.microsoft.com/en-us/azure/storage/files/storage-troubleshoot-linux-file-connection-problems#mount-error13-permission-denied-when-you-mount-an-azure-file-share"
+	fi
+
+	# now try to mount with SMB2.1
+	command="mount -t cifs "$UNCPATH"  "$mountpoint" -o vers=2.1,username="$username",password=$password,dir_mode=0777,file_mode=0777,sec=ntlmssp"
+	print_log "Failed to mount the share with SMB3.0. Try with mounting share using SMB2.1"
+	print_log "$command" "info"
+	sudo sh -c "$command" 2>/tmp/mount.error
+fi
+
+if [[ $? -gt 0 ]];then
+	print_log "Mounting share fails" "error"
+
+	grep "Permission denied" /tmp/mount.error >/dev/null 2>&1
+	if [ $? -eq 0 ]; then
+		#SMB 2.1, provilde hints to Cx
+		echo "The Azure file share failed to mount with mount error (13): Permission denied"
+		echo "This could be due to a number of reasons."
+		echo "To resolve this issue, follow the steps in the Azure Files troubleshooting guide: https://docs.microsoft.com/en-us/azure/storage/files/storage-troubleshoot-linux-file-connection-problems#mount-error13-permission-denied-when-you-mount-an-azure-file-share"
+	fi
+else
+	print_log "Mounting share succeeds" "info"
+fi
```

Comment:
if mount failed for other reasons than permission denied, are we printing those logs to stderr?